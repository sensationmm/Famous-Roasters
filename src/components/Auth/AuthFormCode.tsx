import React, { ChangeEventHandler } from 'react'
import i18n from 'src/config/i18n'

import { AuthFormItemInput } from '.'

interface AuthFormCodeProps {
  screenKey: string
  onChange: ChangeEventHandler<HTMLInputElement>
}

export const AuthFormCode: React.FC<AuthFormCodeProps> = ({ screenKey, onChange }: AuthFormCodeProps) => {
  return (
    <AuthFormItemInput
      name="code"
      label={i18n.t(`auth.${screenKey}.code.label`)}
      rules={[
        { required: true, message: i18n.t(`auth.${screenKey}.code.error.required`) },
        () => ({
          validator(_, value) {
            // eslint-disable-next-line no-useless-escape
            const format = /^\d{6}$/
            if (!value || value.match(format)) {
              return Promise.resolve()
            }
            return Promise.reject(new Error(i18n.t(`auth.${screenKey}.code.error.format`)))
          },
        }),
      ]}
      type="text"
      placeholder={i18n.t(`auth.${screenKey}.code.placeholder`)}
      onChange={onChange}
      dataTestId="code"
    />
  )
}
