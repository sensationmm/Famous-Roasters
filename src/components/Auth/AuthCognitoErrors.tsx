import React from 'react'
import { ErrorBox } from 'src/components'
import i18n from 'src/config/i18n'

interface AuthCognitoErrorsProps {
  errorCode?: string
}

export const AuthCognitoErrors: React.FC<AuthCognitoErrorsProps> = ({ errorCode }: AuthCognitoErrorsProps) => {
  if (errorCode) {
    switch (errorCode) {
      case 'LoginGenericException':
        return (
          <ErrorBox dataTestId="alertLoginGeneric">
            {i18n.t<string>('auth.cognito.errors.loginGeneric.prompt')}{' '}
            <a href="/register" className="border-b">
              {i18n.t<string>('auth.cognito.errors.loginGeneric.ctaText')}.
            </a>
          </ErrorBox>
        )
      case 'ForgotPasswordGenericException':
        return (
          <ErrorBox dataTestId="alertForgotPasswordGeneric">
            {i18n.t<string>('auth.cognito.errors.forgotPasswordGeneric.prompt')}{' '}
            <a href="/register" className="border-b">
              {i18n.t<string>('auth.cognito.errors.forgotPasswordGeneric.ctaText')}.
            </a>
          </ErrorBox>
        )
      case 'RegisterGenericException':
        return (
          <ErrorBox dataTestId="alertRegisterGeneric">
            {i18n.t<string>('auth.cognito.errors.registerGeneric.prompt')}{' '}
            <a href="/login" className="border-b">
              {i18n.t<string>('auth.cognito.errors.registerGeneric.ctaText')}.
            </a>
          </ErrorBox>
        )
      case 'ConfirmUserNotFoundException':
        return (
          <ErrorBox dataTestId="alertConfirmUserNotFound">
            {i18n.t<string>('auth.cognito.errors.confirmUserNotFound.prompt')}{' '}
            <a href="/register" className="border-b">
              {i18n.t<string>('auth.cognito.errors.confirmUserNotFound.ctaText')}.
            </a>
          </ErrorBox>
        )
      case 'ConfirmUserInvalidParameterException':
        return (
          <ErrorBox dataTestId="alertRegisterInvalidParam">
            {i18n.t<string>('auth.cognito.errors.confirmUserInvalidParameter.prompt')}{' '}
            <a href="/login" className="border-b">
              {i18n.t<string>('auth.cognito.errors.confirmUserInvalidParameter.ctaText')}.
            </a>
          </ErrorBox>
        )
      case 'UserNotFoundException':
      case 'UsernameExistsException':
      case 'NotAuthorizedException':
        return (
          <ErrorBox dataTestId="alertIncorrectUserNameOrPassword">
            {i18n.t<string>('auth.cognito.errors.incorrectUserNameOrPassword')}
          </ErrorBox>
        )
      case 'CodeMismatchException':
        return <ErrorBox dataTestId="alertInvalidCode">{i18n.t<string>('auth.cognito.errors.invalidCode')}</ErrorBox>
      default:
        return <ErrorBox dataTestId="alertGeneric">{i18n.t<string>('auth.cognito.errors.generic')}</ErrorBox>
    }
  } else return <></>
}
