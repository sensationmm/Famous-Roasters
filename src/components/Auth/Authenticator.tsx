import { Authenticator as AmplifyAuthenticator } from 'aws-amplify-react'
import { IAuthenticatorProps } from 'aws-amplify-react/lib/Auth/Authenticator'
import React from 'react'

export interface AuthenticatorProps extends IAuthenticatorProps {
  children: React.ReactNode
}

export const Authenticator: React.FC<AuthenticatorProps> = ({ children, ...props }: AuthenticatorProps) => {
  return <AmplifyAuthenticator {...props}>{children}</AmplifyAuthenticator>
}
