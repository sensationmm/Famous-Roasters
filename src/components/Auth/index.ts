import { Authenticator } from './Authenticator'
import { AuthFormEmail } from './AuthFormEmail'
import { AuthFormItemInput } from './AuthFormItemInput'
import { AuthFormPassword } from './AuthFormPassword'
import { AuthSignIn } from './SignIn'

export { Authenticator as Authenticator }
export { AuthFormEmail as AuthFormEmail }
export { AuthFormItemInput as AuthFormItemInput }
export { AuthFormPassword as AuthFormPassword }
export { AuthSignIn as AuthSignIn }
