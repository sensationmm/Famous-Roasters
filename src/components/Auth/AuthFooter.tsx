import React from 'react'
import { Link } from 'react-router-dom'
import { Typography, TypographySize, TypographyType } from 'src/components'
import i18n from 'src/config/i18n'

export const AuthFooter: React.FC = () => {
  return (
    <>
      <div className="flex">
        <div className="border-t border-brand-grey-whisper absolute w-full left-0" />
      </div>
      <div className="w-full mt-4 flex justify-center">
        <Link to="//www.notion.so/Datenschutzhinweise-ad5b45261a964166a74883d2770e28ea" target="_blank">
          <Typography
            as="div"
            type={TypographyType.Paragraph}
            size={TypographySize.Tiny}
            className="font-normal border-b"
          >
            {i18n.t<string>('auth.footer.privacyLink')}
          </Typography>
        </Link>
        <Link
          to="//slender-petunia-c3c.notion.site/Imprint-33b18fd9e469471ab768eef8155da968"
          target="_blank"
          className="ml-10"
        >
          <Typography
            as="div"
            type={TypographyType.Paragraph}
            size={TypographySize.Tiny}
            className="font-normal border-b"
          >
            {i18n.t<string>('auth.footer.imprintLink')}
          </Typography>
        </Link>
      </div>
    </>
  )
}
