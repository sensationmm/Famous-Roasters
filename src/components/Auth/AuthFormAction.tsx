import React from 'react'
import { Typography, TypographySize, TypographyType } from 'src/components'

interface AuthFormActionProps {
  promptText?: string
  onClick: React.MouseEventHandler<HTMLElement>
  dataTestId: string
  ctaText: string
  className?: string
}

export const AuthFormAction: React.FC<AuthFormActionProps> = ({
  promptText,
  onClick,
  dataTestId,
  ctaText,
  className,
}: AuthFormActionProps) => {
  return (
    <Typography as="div" type={TypographyType.Label} size={TypographySize.Tiny} className={className}>
      {promptText?.length && `${promptText}${' '}`}
      <a onClick={onClick} data-testid={dataTestId}>
        {ctaText}
      </a>
    </Typography>
  )
}
