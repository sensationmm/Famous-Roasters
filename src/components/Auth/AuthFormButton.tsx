import React from 'react'
import { Button, ButtonEmphasis, ButtonSize } from 'src/components/Button'

interface AuthFormButtonProps {
  ctaText: string
  disabled?: boolean
  onClick?: () => void
}

export const AuthFormButton: React.FC<AuthFormButtonProps> = ({
  ctaText,
  disabled = false,
  onClick,
}: AuthFormButtonProps) => {
  return (
    <Button
      type="submit"
      emphasis={ButtonEmphasis.Primary}
      size={ButtonSize.lg}
      data-testid="submit"
      className="flex w-full justify-center"
      disabled={disabled}
      onClick={onClick}
    >
      {ctaText}
    </Button>
  )
}
