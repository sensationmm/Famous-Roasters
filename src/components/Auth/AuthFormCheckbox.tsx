import { Field } from 'rc-field-form'
import React, { useState } from 'react'
import { Checkbox } from 'src/components'
import i18n from 'src/config/i18n'

interface AuthFormCheckboxProps {
  screenKey: string
  name: string
  dataTestId?: string
}

export const AuthFormCheckbox: React.FC<AuthFormCheckboxProps> = ({
  screenKey,
  name,
  dataTestId = undefined,
}: AuthFormCheckboxProps) => {
  const [isSelected, setIsSelected] = useState<boolean>(false)

  return (
    <Field
      name={name}
      rules={[
        {
          type: 'string',
          validator: () => {
            return new Promise<void>((resolve, reject) => {
              if (isSelected) {
                resolve()
              } else {
                reject(i18n.t(`auth.${screenKey}.error.required`))
              }
            })
          },
        },
      ]}
    >
      <Checkbox
        dataTestId={dataTestId}
        name={name}
        small={true}
        text={i18n.t<string>(`auth.${screenKey}.label`)}
        selected={isSelected}
        toggleSelected={setIsSelected}
      />
    </Field>
  )
}
