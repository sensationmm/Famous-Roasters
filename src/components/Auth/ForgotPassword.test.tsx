import Auth from '@aws-amplify/auth'
import { act, fireEvent, render, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { MemoryRouter } from 'react-router-dom'
import i18n from 'src/config/i18n'

import { AuthForgotPassword } from '.'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteMatch: () => ({
    url: '/reset-password',
  }),
}))

jest.mock('@aws-amplify/ui-react', () => ({
  ...jest.requireActual('@aws-amplify/ui-react'),
  AmplifyAuthenticator: jest.fn(),
}))

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
delete window.location
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.location = new URL('https://www.60beans.com/reset-password')

describe('ForgotPassword custom auth component', () => {
  const snippet = (initialAuthState = 'forgotPassword', initialAuthData = {}) => (
    <I18nextProvider i18n={i18n}>
      <MemoryRouter initialEntries={['/reset-password']}>
        <AuthForgotPassword authState={initialAuthState} authData={initialAuthData} />
      </MemoryRouter>
    </I18nextProvider>
  )

  it('Renders correctly for an expected state', async () => {
    const { container } = render(snippet())
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly (nothing) for an unexpected state', async () => {
    const { container } = render(snippet('other'))
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can reset his password', async () => {
    const mockForgotPassword = jest.spyOn(Auth, 'forgotPassword')
    mockForgotPassword.mockResolvedValue({ CodeDeliveryDetails: 'something' })
    const mockForgotPasswordSubmit = jest.spyOn(Auth, 'forgotPasswordSubmit')
    mockForgotPasswordSubmit.mockResolvedValue('')
    const { getByTestId } = render(snippet())

    const usernameInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(usernameInput).toBeInTheDocument()
    fireEvent.click(usernameInput)
    fireEvent.change(usernameInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPassword).resolves

    const emailInput = getByTestId('username')
    const codeInput = getByTestId('code')
    const passwordInput = getByTestId('password')
    const passwordRepeatInput = getByTestId('passwordRepeat')
    const submitBtn2 = getByTestId('submit')
    expect(emailInput).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'user@60beans.com' } })
    expect(codeInput).toBeInTheDocument()
    fireEvent.click(codeInput)
    fireEvent.change(codeInput, { target: { value: '123456' } })
    expect(passwordInput).toBeInTheDocument()
    fireEvent.click(passwordInput)
    fireEvent.change(passwordInput, { target: { value: '123456AbC?' } })
    expect(passwordRepeatInput).toBeInTheDocument()
    fireEvent.click(passwordRepeatInput)
    fireEvent.change(passwordRepeatInput, { target: { value: '123456AbC?' } })
    expect(submitBtn2).toBeInTheDocument()
    fireEvent.click(submitBtn2)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPasswordSubmit).resolves

    mockForgotPassword.mockReset()
    mockForgotPasswordSubmit.mockReset()
  })

  it('The user can request a new reset code', async () => {
    const mockForgotPassword = jest.spyOn(Auth, 'forgotPassword')
    mockForgotPassword.mockResolvedValue({ CodeDeliveryDetails: 'something' })
    const { getByTestId } = render(snippet())

    const usernameInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(usernameInput).toBeInTheDocument()
    fireEvent.click(usernameInput)
    fireEvent.change(usernameInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    const emailInput = getByTestId('username')
    const resendLink = getByTestId('resend')
    expect(emailInput).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'user@60beans.com' } })
    expect(resendLink).toBeInTheDocument()
    fireEvent.click(resendLink)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPassword).resolves
    mockForgotPassword.mockReset()
  })

  it('The user reset failing for a not existing user is catched', async () => {
    const mockSignIn2 = jest.spyOn(Auth, 'forgotPassword')
    mockSignIn2.mockRejectedValue(new Error('UserNotFoundException: text'))
    const { getByTestId } = render(snippet())

    const emailInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(emailInput).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    await expect(mockSignIn2).rejects.toThrowError('UserNotFoundException')
  })

  it('The user reset failing for any other reason is catched', async () => {
    const mockForgotPassword = jest.spyOn(Auth, 'forgotPassword')
    mockForgotPassword.mockRejectedValue(new Error('An error'))
    const { getByTestId } = render(snippet())

    const usernameInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(usernameInput).toBeInTheDocument()
    fireEvent.click(usernameInput)
    fireEvent.change(usernameInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPassword).resolves
    mockForgotPassword.mockReset()
  })

  it('The user reset failing when setting the password with an invalid code is catched', async () => {
    const mockForgotPassword = jest.spyOn(Auth, 'forgotPassword')
    mockForgotPassword.mockResolvedValue({ CodeDeliveryDetails: 'something' })
    const mockForgotPasswordSubmit = jest.spyOn(Auth, 'forgotPasswordSubmit')
    mockForgotPasswordSubmit.mockRejectedValue(new Error('CodeMismatchException: text'))
    const { getByTestId } = render(snippet())

    const usernameInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(usernameInput).toBeInTheDocument()
    fireEvent.click(usernameInput)
    fireEvent.change(usernameInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    const emailInput = getByTestId('username')
    const codeInput = getByTestId('code')
    const passwordInput = getByTestId('password')
    const passwordRepeatInput = getByTestId('passwordRepeat')
    const submitBtn2 = getByTestId('submit')
    expect(emailInput).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'user@60beans.com' } })
    expect(codeInput).toBeInTheDocument()
    fireEvent.click(codeInput)
    fireEvent.change(codeInput, { target: { value: '123456' } })
    expect(passwordInput).toBeInTheDocument()
    fireEvent.click(passwordInput)
    fireEvent.change(passwordInput, { target: { value: '123456AbC?' } })
    expect(passwordRepeatInput).toBeInTheDocument()
    fireEvent.click(passwordRepeatInput)
    fireEvent.change(passwordRepeatInput, { target: { value: '123456AbC?' } })
    expect(submitBtn2).toBeInTheDocument()
    fireEvent.click(submitBtn2)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPassword).resolves
    expect(mockForgotPasswordSubmit).resolves
    mockForgotPassword.mockReset()
    mockForgotPasswordSubmit.mockReset()
  })

  it('The user reset failing for wrong code format shows error', async () => {
    const mockForgotPassword = jest.spyOn(Auth, 'forgotPassword')
    mockForgotPassword.mockResolvedValue({ CodeDeliveryDetails: 'something' })
    const { getByTestId } = render(snippet())

    const usernameInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(usernameInput).toBeInTheDocument()
    fireEvent.click(usernameInput)
    fireEvent.change(usernameInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    const emailInput = getByTestId('username')
    const codeInput = getByTestId('code')
    const passwordInput = getByTestId('password')
    const passwordRepeatInput = getByTestId('passwordRepeat')
    const submitBtn2 = getByTestId('submit')
    expect(emailInput).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'user@60beans.com' } })
    expect(codeInput).toBeInTheDocument()
    fireEvent.click(codeInput)
    fireEvent.change(codeInput, { target: { value: 'abc' } })
    expect(passwordInput).toBeInTheDocument()
    fireEvent.click(passwordInput)
    fireEvent.change(passwordInput, { target: { value: '123456AbC?' } })
    expect(passwordRepeatInput).toBeInTheDocument()
    fireEvent.click(passwordRepeatInput)
    fireEvent.change(passwordRepeatInput, { target: { value: '123456AbC?' } })
    expect(submitBtn2).toBeInTheDocument()
    fireEvent.click(submitBtn2)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPassword).resolves
    mockForgotPassword.mockReset()
  })

  it('The user reset failing when setting the password for any other reason is catched', async () => {
    const mockForgotPassword = jest.spyOn(Auth, 'forgotPassword')
    mockForgotPassword.mockResolvedValue({ CodeDeliveryDetails: 'something' })
    const mockForgotPasswordSubmit = jest.spyOn(Auth, 'forgotPasswordSubmit')
    mockForgotPasswordSubmit.mockRejectedValue(new Error('An error'))
    const { getByTestId } = render(snippet())

    const usernameInput = getByTestId('username')
    const submitBtn = getByTestId('submit')
    expect(usernameInput).toBeInTheDocument()
    fireEvent.click(usernameInput)
    fireEvent.change(usernameInput, { target: { value: 'user@60beans.com' } })
    expect(submitBtn).toBeInTheDocument()
    fireEvent.click(submitBtn)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    const emailInput = getByTestId('username')
    const codeInput = getByTestId('code')
    const passwordInput = getByTestId('password')
    const passwordRepeatInput = getByTestId('passwordRepeat')
    const submitBtn2 = getByTestId('submit')
    expect(emailInput).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'user@60beans.com' } })
    expect(codeInput).toBeInTheDocument()
    fireEvent.click(codeInput)
    fireEvent.change(codeInput, { target: { value: '123456' } })
    expect(passwordInput).toBeInTheDocument()
    fireEvent.click(passwordInput)
    fireEvent.change(passwordInput, { target: { value: '123456AbC?' } })
    expect(passwordRepeatInput).toBeInTheDocument()
    fireEvent.click(passwordRepeatInput)
    fireEvent.change(passwordRepeatInput, { target: { value: '123456AbC?' } })
    expect(submitBtn2).toBeInTheDocument()
    fireEvent.click(submitBtn2)

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 500))
    })

    expect(mockForgotPassword).resolves
    expect(mockForgotPasswordSubmit).resolves
    mockForgotPassword.mockReset()
    mockForgotPasswordSubmit.mockReset()
  })

  it('Shows incorrect email error', async () => {
    const { getByTestId } = render(snippet('forgotPasswordError', { errorCode: 'ForgotPasswordGenericException' }))

    await waitFor(() => {
      const error = getByTestId('alertForgotPasswordGeneric')
      expect(error).toBeInTheDocument()
    })
  })

  it('Shows incorrect code error', async () => {
    const { getByTestId } = render(snippet('forgotPasswordError', { errorCode: 'CodeMismatchException' }))

    await waitFor(() => {
      const error = getByTestId('alertInvalidCode')
      expect(error).toBeInTheDocument()
    })
  })

  it('Shows generic error', async () => {
    const { getByTestId } = render(snippet('forgotPasswordError', { errorCode: 'OtherErrorCode' }))

    await waitFor(() => {
      const error = getByTestId('alertGeneric')
      expect(error).toBeInTheDocument()
    })
  })

  it('The user can navigate to login', async () => {
    const { getByTestId } = render(snippet())

    await waitFor(() => {
      const goToSignInLink = getByTestId('goToSignInLink')
      expect(goToSignInLink).toBeInTheDocument()
      fireEvent.click(goToSignInLink)
    })
  })
})
