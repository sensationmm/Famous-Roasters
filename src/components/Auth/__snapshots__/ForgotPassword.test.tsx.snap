// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword custom auth component Renders correctly (nothing) for an unexpected state 1`] = `
<div>
  <div
    class="my-4 mx-6 md:mx-0"
  >
    <div
      class="flex w-full"
    >
      <span
        class="text-2xl leading-8 font-semibold"
      >
        Passwort zur端cksetzen
      </span>
    </div>
    <div
      class="flex w-full mt-2"
    >
      <span
        class="text-sm leading-5"
      >
        Kein Problem. Gib deine E-Mail-Adresse ein, mit der du ein Konto bei 60beans erstellt hast und wir schicken dir einen neuen Link zur Passworterstellung zu.
      </span>
    </div>
    <div
      class="mt-4"
    />
    <form>
      <div
        class="w-full mt-8"
      >
        <label
          class="relative"
        >
          <span
            class="absolute -top-6 left-3 px-1 bg-white"
          >
            <span
              class="text-sm leading-5"
            >
              Deine E-mail-Adresse
            </span>
          </span>
          <input
            class="w-full border rounded-full px-4 py-2 focus:outline-0 focus:ring focus:ring-primary/30"
            data-testid="username"
            placeholder="Deine E-mail-Adresse"
            type="email"
            value=""
          />
        </label>
      </div>
      <div
        class="mt-8"
      >
        <button
          class="flex w-full justify-center relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-md font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary pointer-events-none opacity-70"
          data-testid="submit"
          disabled=""
          role="button"
          type="submit"
        >
          Schicken
        </button>
      </div>
      <div
        class="flex w-fit mt-6 mb-6 border-b cursor-pointer text-xs leading-4 font-semibold"
      >
        <a
          data-testid="goToSignInLink"
        >
          Zur端ck zur Anmeldung
        </a>
      </div>
    </form>
    <div
      class="mt-12"
    >
      <div
        class="flex"
      >
        <div
          class="border-t border-brand-grey-whisper absolute w-full left-0"
        />
      </div>
      <div
        class="mt-4 mb-4 text-xl leading-7 font-semibold"
      >
        Ich bin neu hier
      </div>
      <button
        class="flex w-full justify-center relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-md font-bold text-brand-black bg-white border-brand-black hover:border-coreUI-text-tertiary active:border-brand-black"
        data-testid="goToSignUpLink"
        role="button"
        type="button"
      >
        Registrieren
      </button>
    </div>
    <div
      class="mt-12"
    >
      <div
        class="flex"
      >
        <div
          class="border-t border-brand-grey-whisper absolute w-full left-0"
        />
      </div>
      <div
        class="w-full mt-4 flex justify-center"
      >
        <a
          href="//www.notion.so/Datenschutzhinweise-ad5b45261a964166a74883d2770e28ea"
          target="_blank"
        >
          <div
            class="font-normal border-b text-xs leading-4"
          >
            Datenschutz-Bestimmungen
          </div>
        </a>
        <a
          class="ml-10"
          href="//slender-petunia-c3c.notion.site/Imprint-33b18fd9e469471ab768eef8155da968"
          target="_blank"
        >
          <div
            class="font-normal border-b text-xs leading-4"
          >
            Impressum
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ForgotPassword custom auth component Renders correctly for an expected state 1`] = `
<div>
  <div
    class="my-4 mx-6 md:mx-0"
  >
    <div
      class="flex w-full"
    >
      <span
        class="text-2xl leading-8 font-semibold"
      >
        Passwort zur端cksetzen
      </span>
    </div>
    <div
      class="flex w-full mt-2"
    >
      <span
        class="text-sm leading-5"
      >
        Kein Problem. Gib deine E-Mail-Adresse ein, mit der du ein Konto bei 60beans erstellt hast und wir schicken dir einen neuen Link zur Passworterstellung zu.
      </span>
    </div>
    <div
      class="mt-4"
    />
    <form>
      <div
        class="w-full mt-8"
      >
        <label
          class="relative"
        >
          <span
            class="absolute -top-6 left-3 px-1 bg-white"
          >
            <span
              class="text-sm leading-5"
            >
              Deine E-mail-Adresse
            </span>
          </span>
          <input
            class="w-full border rounded-full px-4 py-2 focus:outline-0 focus:ring focus:ring-primary/30"
            data-testid="username"
            placeholder="Deine E-mail-Adresse"
            type="email"
            value=""
          />
        </label>
      </div>
      <div
        class="mt-8"
      >
        <button
          class="flex w-full justify-center relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-md font-bold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary pointer-events-none opacity-70"
          data-testid="submit"
          disabled=""
          role="button"
          type="submit"
        >
          Schicken
        </button>
      </div>
      <div
        class="flex w-fit mt-6 mb-6 border-b cursor-pointer text-xs leading-4 font-semibold"
      >
        <a
          data-testid="goToSignInLink"
        >
          Zur端ck zur Anmeldung
        </a>
      </div>
    </form>
    <div
      class="mt-12"
    >
      <div
        class="flex"
      >
        <div
          class="border-t border-brand-grey-whisper absolute w-full left-0"
        />
      </div>
      <div
        class="mt-4 mb-4 text-xl leading-7 font-semibold"
      >
        Ich bin neu hier
      </div>
      <button
        class="flex w-full justify-center relative inline-flex flex-row items-center rounded-full border-2 px-4 py-2 text-md font-bold text-brand-black bg-white border-brand-black hover:border-coreUI-text-tertiary active:border-brand-black"
        data-testid="goToSignUpLink"
        role="button"
        type="button"
      >
        Registrieren
      </button>
    </div>
    <div
      class="mt-12"
    >
      <div
        class="flex"
      >
        <div
          class="border-t border-brand-grey-whisper absolute w-full left-0"
        />
      </div>
      <div
        class="w-full mt-4 flex justify-center"
      >
        <a
          href="//www.notion.so/Datenschutzhinweise-ad5b45261a964166a74883d2770e28ea"
          target="_blank"
        >
          <div
            class="font-normal border-b text-xs leading-4"
          >
            Datenschutz-Bestimmungen
          </div>
        </a>
        <a
          class="ml-10"
          href="//slender-petunia-c3c.notion.site/Imprint-33b18fd9e469471ab768eef8155da968"
          target="_blank"
        >
          <div
            class="font-normal border-b text-xs leading-4"
          >
            Impressum
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
